### ISSUE TITLE
Duplicate H element rule might force duplication

### ISSUE THREAD

**Issue by trygve-lie** [CSSLint/csslint#80]
The following style:

h1, h2 {font-face: Verdana;}
h1 {font-size: 2.1em;}
h2 {font-size: 1.8em;}

does prompt a warning due to the recommendation that one should only define H elements only once in a style. Tough; In this example one would have to replicate the font-face for all H tags to conform with the lint rule. That would increase the size of the css file.

Isn't it most common to set some styles which apply to all H elements and then override only the differences, like the font-size, for each H element?

**Comment by bronson** [comment]
I don't quite understand the conclusion...  People are agreeing that the current rule is quite specialized, but should be left on by default?  That seems like a recipe for false positives.

My case: I'm styling Markdown-generated HTML so headings can't nest.  Here's my markup:

```css
h1,h2,h3,h4,h5,h6 {
  font-family: 'Lucida Grande', Helvetica, Verdana, sans-serif;
  font-weight: bold;
}
h1,h2 { color: rgb(52,101,164); }
h3,h4,h5,h6 { color: rgb(114,159,207); }
h1 { margin-top: 1.3em; font-size: 1.5em; }
h2 { font-size: 1.3em; }
```

Is the linter suggesting that there's a better way for me to write this?

**Comment by Nateowami** [comment]
Pardon me if I'm wrong, but it appears to me that this rule—and most of the discussion of it—even its documentation—misses [the entire point of the rule](https://github.com/CSSLint/csslint/wiki/Headings-should-only-be-defined-once):
> The heading elements (h1-h6) are considered to be built-in objects that should look the same regardless of where they appear.

The documentation gives this example of "bad code":
``` css
/* Two rules for h3 */
h3 {
    font-weight: normal;
}

.box h3 {
    font-weight: bold;
}
```
But if the idea is to make heading elements appear the same anywhere in the page, then it takes only one rule to break this:
``` css
.box h3 {
    /* There. Now h3 inside of a .box and outside will differ. With only one rule 
    declared. The h3 is no longer atomic, but no warnings are shown. */
    font-weight: bold;
}
```
Even this code follows the spirit of the rule:
``` css
h1, h2 {
    color: black;
    font-weight: bold;
}
h1 {
  font-size: 48pt;
}
/* There. Now h1 and h2 will look the same no matter where on the page. 
   The h1 and h2 headings are still atomic, but the linter complains about multiple 
   rules. */
```
It makes *no difference whatsoever* whether there is one rule or a thousand rules. What matters is whether the headings are qualified, making them context-dependent, and therefore non-atomic.

In fact, I cannot think of *a single instance* where this rule would catch non-atomic headings that wouldn't be caught by a "don't qualify headers rule" (Of course things like `h3:hover` need to be allowed, but that shouldn't be a problem).

Am I missing something? Is there a legitimate reason to count the number of rules instead of whether rules are qualified?
